L1Trigger/L1THGCal/plugins/geometries/HGCalTriggerGeometryV9Imp1.cc

```diff
@@ -0,0 +1,1100 @@
+        unsigned cellu = cell_det_id.cellU();
+        unsigned cellv = cell_det_id.cellV();
+        auto trigger_cell_itr = cells_to_trigger_cells_.find(packCellId(type, cellu, cellv));
+        if(trigger_cell_itr==cells_to_trigger_cells_.end())
+        {
+            throw cms::Exception("BadGeometry")
+                << "HGCalTriggerGeometry: HGCal  cell " << cellu << "," << cellv << " in wafer type "<<type<<" is not mapped to any trigger cell. The trigger cell mapping should be modified.\n";
+        }
+        auto wafer_trigger_cell_itr = wafers_to_wafers_old_.find(packWaferId(waferu, waferv));
+        if(wafer_trigger_cell_itr==wafers_to_wafers_old_.end())
+        {
+            throw cms::Exception("BadGeometry")
+                << "HGCalTriggerGeometry: Wafer "<<waferu<<","<<waferv<<" is not mapped to any trigger wafer ID. The wafer mapping should be modified.\n";
+        }
+        trigger_cell = trigger_cell_itr->second;
+        wafer_trigger_cell = wafer_trigger_cell_itr->second;
+    }
+    return HGCalDetId((ForwardSubdetector)subdet, zside, layer, tc_type, wafer_trigger_cell, trigger_cell).rawId();
+
+}
+
+unsigned 
+HGCalTriggerGeometryV9Imp1::
+getModuleFromCell( const unsigned cell_id ) const
+{
+    return getModuleFromTriggerCell(getTriggerCellFromCell(cell_id));
+}
+
+unsigned 
+HGCalTriggerGeometryV9Imp1::
+getModuleFromTriggerCell( const unsigned trigger_cell_id ) const
+{
+    HGCalDetId trigger_cell_det_id(trigger_cell_id);
+    unsigned module = 0;
+    // Scintillator
+    if(trigger_cell_det_id.subdetId()==ForwardSubdetector::HGCHEB)
+    {
+        // For scintillator, the module ID is currently encoded as the wafer in HGCalDetId
+        module = trigger_cell_det_id.wafer();
+    }
+    // Silicon
+    else
+    {
+        auto module_itr = wafer_to_module_.find(trigger_cell_det_id.wafer());
+        if(module_itr==wafer_to_module_.end())
+        {
+            throw cms::Exception("BadGeometry")
+                <<trigger_cell_det_id
+                << "HGCalTriggerGeometry: Wafer " << trigger_cell_det_id.wafer() << " is not mapped to any trigger module. The module mapping should be modified. See https://twiki.cern.ch/twiki/bin/viewauth/CMS/HGCALTriggerPrimitivesSimulation#Trigger_geometry for details.\n";
+        }
+        module = module_itr->second;
+    }
+    return HGCalDetId((ForwardSubdetector)trigger_cell_det_id.subdetId(), trigger_cell_det_id.zside(), trigger_cell_det_id.layer(), trigger_cell_det_id.waferType(), module, HGCalDetId::kHGCalCellMask).rawId();
+}
+
+HGCalTriggerGeometryBase::geom_set 
+HGCalTriggerGeometryV9Imp1::
+getCellsFromTriggerCell( const unsigned trigger_cell_id ) const
+{
+    HGCalDetId trigger_cell_det_id(trigger_cell_id);
+    geom_set cell_det_ids;
+    unsigned subdet = trigger_cell_det_id.subdetId();
+    // Scintillator
+    if(subdet==ForwardSubdetector::HGCHEB)
+    {
+        unsigned trigger_wafer = trigger_cell_det_id.wafer();
```

Comment:
could reduce code duplication by putting the common variables from `trigger_cell_det_id` outside of the conditional blocks

Reply:
could also make one of the "unpack" functions to provide the `trigger_cell_det_id` quantities